presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -73,19 +68,29 @@
         implements ConnectorSplitManager
 {
     private static final Logger log = Logger.get(HudiSplitManager.class);
+
     private final HdfsEnvironment hdfsEnvironment;
     private final HudiTransactionManager hudiTransactionManager;
     private final HudiPartitionManager hudiPartitionManager;
+    private final ExecutorService asyncQueueExecutor;
+    private final ScheduledExecutorService splitLoaderExecutorService;
+    private final ExecutorService splitGeneratorExecutorService;
 
     @Inject
     public HudiSplitManager(
             HdfsEnvironment hdfsEnvironment,
             HudiTransactionManager hudiTransactionManager,
-            HudiPartitionManager hudiPartitionManager)
+            HudiPartitionManager hudiPartitionManager,
+            @ForHudiSplitAsyncQueue ExecutorService asyncQueueExecutor,
+            @ForHudiSplitSource ScheduledExecutorService splitLoaderExecutorService,
+            @ForHudiBackgroundSplitLoader ExecutorService splitGeneratorExecutorService)
```

Comment:
A bit complex. Maybe we don't need three executor services here.

Reply:
I wrote the async split loader  for Hudi Connector of Trino. It might be a good reference. 

https://github.com/trinodb/trino/pull/10228/files#diff-fe90cd79555894027a34ea1321a7573faa127ffa9c5a64c241c2d4182018c1ac

Reply:
I think it makes sense to have separate executors for the split loader and split generator so that the two can work independently in parallel.

Reply:
To @7c00 's point, would this cause more context switch though with too many threads. Love to follow on here. and understand the differences.

Reply:
In case of Trino, both split loader and partition listing share the same executor service. Here, we created separate as both have different purpose. One is simply adding partitions to the async queue, the other is listing those partitions (IO can be unpredictive).